tests/coverage/coverage.cpp

```diff
@@ -267,10 +267,67 @@ TEST(ImporterCoverage, importingComponentWithCnUnitsThatAreEmpty)
     model->addComponent(c);
 
     EXPECT_FALSE(model->hasUnresolvedImports());
+
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(4), validator->errorCount());
+
     model = importer->flattenModel(model);
 
+    validator->validateModel(model);
+    EXPECT_EQ(size_t(1), validator->errorCount());
+}
+
+TEST(ImporterCoverage, importingComponentWithCnUnitsThatReferenceEmptyNamedUnits)
+{
+    const std::string in =
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+        "<model xmlns=\"http://www.cellml.org/cellml/2.0#\" name=\"myModel\">\n"
+        "  <units name=\"\">\n"
+        "    <unit units=\"metre\" prefix=\"milli\"/>\n"
+        "  </units>\n"
+        "   <component name=\"myComponent\">\n"
+        "       <variable name=\"a\" units=\"\"/>\n"
+        "       <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
+        "           <apply>"
+        "             <eq/>\n"
+        "             <ci>a</ci>\n"
+        "             <cn cellml:unuts=\"\">1</cn>\n"
+        "           </apply>\n"
+        "       </math>\n"
+        "   </component>\n"
+        "</model>";
+
+    // Create the model by parsing the string above.
+    auto parser = libcellml::Parser::create();
+    auto importedModel = parser->parseModel(in);
+    auto importer = libcellml::Importer::create();
+    auto validator = libcellml::Validator::create();
+
+    auto model = libcellml::Model::create("myModel");
+    auto u = libcellml::Units::create("mm");
+    u->addUnit("metre", "milli");
+    model->addUnits(u);
+
+    auto c = libcellml::Component::create("c");
+
+    auto importSource = libcellml::ImportSource::create();
+    importSource->setUrl("not_required_resolving_import_manually");
+    importSource->setModel(importedModel);
+
+    c->setImportReference("myComponent");
+    c->setImportSource(importSource);
+    model->addComponent(c);
+
+    EXPECT_FALSE(model->hasUnresolvedImports());
+
     validator->validateModel(model);
     EXPECT_EQ(size_t(3), validator->errorCount());
```

Comment:
Again, I wish we could see the kind of errors we are expecting here and at line 330.

Reply:
[d722583](https://github.com/cellml/libcellml/pull/1167/commits/d722583a6899b10ea5aa8f0bdcd5353f0177e056)